src/components/Timelog/WeeklySummaries.jsx

```diff
@@ -1,23 +1,40 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect} from 'react';
 import parse from 'html-react-parser';
 import './Timelog.css'
-import updateWeeklySummaries from 'actions/weeklySummaries';
+import {updateWeeklySummaries} from '../../actions/weeklySummaries';
 import { getUserProfile, updateUserProfile } from 'actions/userProfile';
 import hasPermission from 'utils/permissions';
 import { connect, useDispatch, useSelector } from 'react-redux';
 import { Editor } from '@tinymce/tinymce-react';
 import { userProfileByIdReducer } from 'reducers/userProfileByIdReducer';
+import Spinner from 'react-bootstrap/Spinner';
 
 const WeeklySummaries = ({ userProfile }) => {
+
+  useEffect(() => {
+      setEditedSummaries([
+        userProfile.weeklySummaries[0]?.summary || '',
+        userProfile.weeklySummaries[1]?.summary || '',
+        userProfile.weeklySummaries[2]?.summary || '',
+      ]);
+    
+  }, [userProfile]);
+
   // Initialize state variables for editing and original summaries
+  
   const [editing, setEditing] = useState([false, false, false]);
+
   const [editedSummaries, setEditedSummaries] = useState([
     userProfile.weeklySummaries[0]?.summary || '',
     userProfile.weeklySummaries[1]?.summary || '',
     userProfile.weeklySummaries[2]?.summary || '',
   ]);
   const [originalSummaries, setOriginalSummaries] = useState([...editedSummaries]);
 
+  const [LoadingHandleSave, setLoadingHandleSave] = useState(null);
```

Comment:
// Consider refactoring to use camelCase for consistency everywhere in the codebase.
const [loadingHandleSave, setLoadingHandleSave] = useState(null);